### DISCUSSION TITLE
can't deploy after npm run dev on local

### DISCUSSION THREAD

**Discussion by fredbouin** [statamic/cms#4511] [Troubleshooting] [answered]
If I work locally and do a ```npm run dev``` and then push my changes to my repo I get the below error message in forge when deploying.

```
error: Your local changes to the following files would be overwritten by merge:
	public/css/tailwind.css
	public/js/site.js
	public/mix-manifest.json
Please commit your changes or stash them before you merge.
```

my deploy script looks like this: 

```
cd /home/forge/mysite.com
git pull origin master
npm install && npm run production
php please cache:clear
```

My typical work flow to push to my repo is:
```
git add .
git commit -m "changed some stuff"
git push origin master
```

**Comment by jasonvarga** [comment, marked as answer]
If you're compiling on your server, then you shouldn't version control those files in your `public` directory.

`git rm public/css/tailwind.css public/js/site.js public/mix-manifest.json`

**Reply by fredbouin** [reply]
Hey Jason, thanks for the response, I'm still learning git so there are still things I am not 100% on. I sort of came to that conclusion as I was troubleshooting but tried to solved it by adding those files to gitignore, but it didn't work, would that not be the same as doing git rm?

I ended up fixing my problem by removing run production for my deployment script and running it locally before pushing changes, is there a down side to this method.

**Reply by jasonvarga** [reply]
If a file has already been tracked by git, adding it to gitignore isn't enough. You need to untrack it (which is what the `rm` command would do).

**Reply by fredbouin** [reply]
Got it, so with files ignored and removed from git, I will be able to npm run dev to compile on my development environment and then npm run production on the server for deployment?

**Reply by jasonvarga** [reply]
Yes, that's the idea!

When you deploy, since Git no longer cares about those 3 files from you original post, they wouldn't be deployed, and you won't get that merge conflict.
